### DISCUSSION TITLE
How to SetSide only on the outside of the domain

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#18376] [Q&A General] [answered]
Hi all,

Just wondering on how to How to SetSide only on the outside of the domain?

This is what I use

```
  [add_side_sets]
    type = SideSetsFromNormalsGenerator
    input = mesh
    normals = '0  0  1
               0  1  0
               1  0  0
               0  0 -1
               0 -1  0
              -1  0  0'
    fixed_normal = true
    new_boundary = 'front top right back bottom left'
    variance = 0.5
    #
  []
```

![444](https://user-images.githubusercontent.com/79562439/126167328-aeb2953b-390d-4d4e-ab2f-9c2c1d2df4e8.PNG)

I don't want the middle planes to be considered as a side, does anyone know how to fix this?

Thank you!

Kind regards,
Traiwit

**Comment by GiudGiud** [comment, marked as answer]
Hello

Are these inner sidesets perfectly horizontal? If not, you could tighten down the variance. 
If so, I would use another sideset generator for the horizontal sides. 
Like a [ParsedSideSetGenerator](https://mooseframework.inl.gov/source/meshgenerators/ParsedGenerateSideset.html) with `z < eps` or a [SidesetAroundSubdomainGenerator](https://mooseframework.inl.gov/source/meshgenerators/SideSetsAroundSubdomainGenerator.html) , which will also accept a normal and should stick to external faces. 

Guillaume

**Reply by Traiwit** [reply]
awesome, got it thanks @GiudGiud 

a quick question, can the function take min/max value? for example 'z = max'

**Reply by GiudGiud** [reply]
I would avoid an equal, because this could fail with numerical precision errors. But I think that would be parsed correctly
